### ISSUE TITLE
Prefer boolean prop usage at first

### ISSUE THREAD

**Issue by jd-solanki** [vuejs/eslint-plugin-vue#1870]
**Please describe what the rule should do:**
This rule is related to this [rule](https://eslint.vuejs.org/rules/prefer-prop-type-boolean-first.html#vue-prefer-prop-type-boolean-first). In the existing rule we are enforcing defining the boolean rules first but in this rule, we can enforce using boolean props first.

**What category should the rule belong to?**

[x] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
   <!-- GOOD -->
    <v-btn
      flat
      color="success"
    >
      Success
    </v-btn>

   <!-- BAD -->
    <v-btn
      color="info"
      flat
    >
      Info
    </v-btn>

   <!-- GOOD -->
    <v-btn
      flat
      variant="outilned"
      color="success"
      rounded="pill"
    >
      Success
    </v-btn>
```

**Additional context**
Placing boolean prop usage on top makes it easy to read the component props. 👀

**Comment by FloEdelmann** [comment]
That would collide with the [Style Guide](https://vuejs.org/style-guide/rules-recommended.html#element-attribute-order) and the existing [vue/attributes-order](https://eslint.vuejs.org/rules/attributes-order.html) rule.

Maybe it would be possible to add a new option `SHORTHAND_BOOL_ATTR` to that rule. But the logic should not change for existing configurations:

* If `SHORTHAND_BOOL_ATTR` is present in the config, `OTHER_ATTR` would only apply to non-shorthand attributes.
* If `SHORTHAND_BOOL_ATTR` is not present in the config, `OTHER_ATTR` would apply to both shorthand and non-shorthand attributes (like it is now).

**Comment by jd-0001** [comment]
`SHORTHAND_BOOL_ATTR` looks good. ❤️

**Comment by FloEdelmann** [comment]
See also #1728. That issue should probably be implemented together with this one.
